docs/modules/kubernetes/pages/kubernetes-persistence.adoc

```diff
@@ -0,0 +1,42 @@
+ - When scaling up or down, Hazelcast automatically switches to `ACTIVE` cluster state, so partitions are rebalanced and data is spread across all members.
+
+TIP: For a step-by-step tutorial that details how to deploy Hazelcast with persistence enabled, see https://docs.hazelcast.com/tutorials/hazelcast-platform-operator-external-backup-restore[Restore a Cluster from Cloud Storage with Hazelcast Platform Operator]
+
+== Requirements
+Automatic cluster state management requires:
+
+- Hazelcast configured with xref:storage:configuring-persistence.adoc[persistence] enabled
+- xref:kubernetes:deploying-in-kubernetes.adoc[Kubernetes discovery] is in use, either explicitly configured or as auto-detected join configuration
+- Hazelcast is deployed in a `StatefulSet`
+- Hazelcast is executed with a cluster role that is allowed access to `apps` Kubernetes API group and `statefulsets` resources with the `watch` verb. See https://raw.githubusercontent.com/hazelcast/hazelcast/master/kubernetes-rbac.yaml[the proposed `ClusterRole` configuration]
+
+== Configuration
+
+Automatic cluster state management is enabled by default when Hazelcast is configured with persistence enabled and Kubernetes discovery. It can be explicitly disabled by setting the `hazelcast.persistence.auto.cluster.state` Hazelcast property to `false`.
+
+Depending on the use case, the `hazelcast.persistence.auto.cluster.state.strategy` Hazelcast property configures which cluster state will be used when members are temporarily missing from the cluster, e.g., pod is rescheduled or rolling restart is in progress. This property has the following values:
+
+ - `NO_MIGRATION`: Use this as the cluster state if the cluster hosts a mix of persistent and in-memory data structures or if the cluster hosts only persistent data, but you favour availability over speed of recovery. While a member is missing from the cluster, cluster state switches to `NO_MIGRATION`: this way, the first replica of partitions owned by the missing member are promoted and the data are available. When the member rejoins the cluster, persistent data can be recovered from disk and a differential sync (assuming Merkle trees are enabled, which is the case by default for persistent `IMap` and `ICache`) brings them up to speed. For in-memory data structures, a full partition sync is required. This is the default value.
+ - `FROZEN`: Use this cluster state, if your cluster hosts only persistent data structures and you do not mind temporarily losing availability of partitions owned by a missing member in exchange for a speedy recovery from disk. Once the member rejoins, no sync over the network is required.
+
+== Best practices
+
+When running Hazelcast with persistence in Kubernetes, the following configuration is recommended:
+
+ - Use `/hazelcast/health/node-state` as liveness probe and `/hazelcast/health/ready` as readiness probe.
```

Comment:
liveness? not liveliness? I'm sure the term is liveliness.....

Reply:
Was ready to change that but actually, does look right according to https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/